File: Java/Car.java
class Car {
    private Integer id;
    private String license;
    private Account driver;
    // En esta parte del Account
    private Integer passenger;

    public Car(String license, Account name) {
        /*
         * Aqui tambien podemos ver que las variables de arriba son las que se piden
         * segun su
         * tipo de dato y abajo se les designa a que valor iran, es similar a JS
         */
        this.license = license;
        this.driver = name;
        // El poner el nombre del atributo en la super clase de la siguiente forma
        // this.atributo = atributo
        /*
         * Dicho esto vamos construyendo el metodo constructor teniendo cuidado con
         * tener
         * consistencia de los tipos de datos
         */
    }

    void printDataCar() {
        if (passenger != null) {
            System.out.println("License: " + license + " name Driver: " + driver.name + " Passengers = " + passenger);
            // Aqui al actualizar el metodo de Car debemos poner que en lugar de drive se
            // imprima driver.name
        } else {

        }
    }

    // Creando el getter, el cual tiene como devolucion entero pues es lo que va a
    // enviar
    public Integer getPassenger() {
        return passenger;
    }

    // Creando el setter el cual no devuelve nada por lo que se deja este como void
    // y tiene como parametro la variable passenger
    public void setPassenger(Integer passenger) {
        //33 Estamos viendo el polimorfismo asi que vamos a ir viendo como esta funcion no tendra el mismo comportamiento en uberVan
        // Aqui agragamos un if para validar que se ingrese el valor que queremos
        if (passenger == 4) {
            this.passenger = passenger;
        } else {
            System.out.println("Necesitas poner 4 pasajeros");
        }
    }

    // Aqui dimos click derecho para abrir el menu, nos fuimos a acciones del codigo
    // fuente y generate getter and setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getLicense() {
        return license;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public Account getDriver() {
        return driver;
    }

    public void setDriver(Account driver) {
        this.driver = driver;
    }

}

File: PHP/Car.php
<?php
//Invocamos la clase account para tener sus cosas disponibles
require_once('account.php');
class car
{
  public $id;
  public $license;
  public $driver;
  protected $passenger;

  //Metodo constructor
  public function __construct($license, $driver)
  {
    $this->license = $license;
    $this->driver = $driver;
  }
  //Forma de poner salto de linea en PHP tranformando la palabra reservada \n a etiquetas br de HTMl
  public function lineJump()
  {
    $text = "\n";
    echo nl2br($text);
  }

  /*Aqui podemos ver como se declararia
      una funcion en PHP*/

  public function printDataCar()
  {
    /*echo "license: $this->license, 
        conductor: {$this->driver->name}, document: {$this->driver->document}";*/
    if ($this->passenger != null) {
      echo "--> Licencia: $this->license Driver: " . $this->driver->name;
    } else {
    }
    //Es bastante importante entender lo que esta pasando en cuanto a la impresion de los datos del auto siendo que debemos fijarnos en la posicion de las comillas ademas debemos ver que las llaves deben deberse a que estamos accediendo al dato name del driver y lo mismo con document
  }

  public function getPassenger()
  {
    return $this->passenger;
  }

  public function setPassenger($passenger)
  {

    if ($passenger == 4) {
      $this->passenger = $passenger;
    } else {
      echo 'Necesitas asignar 4 pasajeros';
    }
  }

  //void printDataCar() {
  //System.out.println("License: " + license + "name Driver: " + driver.name);
  //Aqui al actualizar el metodo de Car debemos poner que en lugar de drive se imprima driver.name
  // }
}
